lib/http/uri_normalizer.rb

```diff
@@ -0,0 +1,25 @@
+require "uri"
+
+module HTTP
+  module URINormalizer
+    module_function
+
+    # Normalize URI
+    #
+    # @param [#to_s] uri
+    # @return [::URI]
+    def normalize_uri(uri)
```

Comment:
Although Addressable is absolutely awesome gem, it's about 1.7x slower on parsing URIs.